internal/ingest/ingest.go

```diff
@@ -389,6 +391,9 @@ func removeProcessedFrozen(ctx context.Context, dstore datastore.Datastore) erro
 // The Context argument controls the lifetime of the sync. Canceling it cancels
 // the sync and causes the multihash channel to close without any data.
 func (ing *Ingester) Sync(ctx context.Context, peerInfo peer.AddrInfo, depth int, resync bool) (cid.Cid, error) {
+	ctx, s := telemetry.StartSpan(ctx, "Ingester.Sync")
+	defer s.End()
+
```

Comment:
Do you think it would be worth capturing any errors here via a named return? - e.g.
```suggestion
func (ing *Ingester) Sync(ctx context.Context, peerInfo peer.AddrInfo, depth int, resync bool) (_ cid.Cid, err error) {
  ctx, s := telemetry.StartSpan(ctx, "Ingester.Sync")
  defer func() {
    if err != nil {
      telemetry.Error(span, err, "sync failed or something")
    }
    s.End()
  }()
```